messaging/src/main/java/org/axonframework/commandhandling/gateway/ReactiveCommandGateway.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.commandhandling.CommandBus;
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MetaData;
+import org.axonframework.messaging.ReactiveMessageDispatchInterceptorSupport;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Function;
+
+/**
+ * Variation of {@link CommandGateway}. Provides support for reactive return type such as {@link Mono} from Project
+ * Reactor.
+ *
+ * @author Milan Savic
+ * @since 4.4
+ */
+public interface ReactiveCommandGateway extends ReactiveMessageDispatchInterceptorSupport<CommandMessage<?>> {
```

Comment:
Please also have it implement MessageDispatchInterceptorSupport. Preferably, I'd like the Reactive implementations to show more compatibility by extending the MessageDispatchInterceptorSupport. The reactive impl could do an instanceof check to see if implementations are already reactive. If not, it could wrap them with a reactive wrapper implementation.